lib/cocoapods/installer/xcode/pods_project_generator/app_host_installer.rb

```diff
@@ -0,0 +1,134 @@
+            @sandbox = sandbox
+            @project = project
+            @platform = platform
+            @test_type = test_type
+          end
+
+          # @return [PBXNativeTarget] the app host native target that was installed.
+          #
+          def install!
+            name = app_host_label
+            platform_name = platform.name
+            app_host_target = Pod::Generator::AppTargetHelper.add_app_target(project, platform_name, deployment_target,
+                                                                             name)
+            app_host_target.build_configurations.each do |configuration|
+              configuration.build_settings['PRODUCT_NAME'] = name
+              configuration.build_settings['PRODUCT_BUNDLE_IDENTIFIER'] = 'org.cocoapods.${PRODUCT_NAME:rfc1034identifier}'
+              configuration.build_settings['CODE_SIGN_IDENTITY'] = '' if platform == :osx
+              configuration.build_settings['CURRENT_PROJECT_VERSION'] = '1'
+            end
+            Pod::Generator::AppTargetHelper.add_app_host_main_file(project, app_host_target, platform_name, name)
+            create_info_plist_file(app_host_info_plist_path, app_host_target, '1.0.0', platform, :appl)
+            project[name].new_file(app_host_info_plist_path)
+            app_host_target
+          end
+
+          private
+
+          # @return [Pathname] The absolute path of the Info.plist to use for an app host.
+          #
+          def app_host_info_plist_path
+            project.path.dirname.+("#{app_host_label}/#{app_host_label}-Info.plist")
+          end
+
+          # @return [String] The label of the app host label to use given the platform and test type.
+          #
+          def app_host_label
+            "AppHost-#{Platform.string_name(platform.symbolic_name)}-#{test_type.capitalize}-Tests"
+          end
+
+          # @return [String] The deployment target.
+          #
+          def deployment_target
+            platform.deployment_target.to_s
+          end
+
+          # Creates the Info.plist file which sets public framework attributes
+          #
+          # @param  [Pathname] path
+          #         the path to save the generated Info.plist file.
+          #
+          # @param  [PBXNativeTarget] native_target
+          #         the native target to link the generated Info.plist file into.
+          #
+          # @param  [Version] version
+          #         the version to use for when generating this Info.plist file.
+          #
+          # @param  [Platform] platform
+          #         the platform to use for when generating this Info.plist file.
+          #
+          # @param  [Symbol] bundle_package_type
+          #         the CFBundlePackageType of the target this Info.plist file is for.
+          #
+          # @return [void]
+          #
+          def create_info_plist_file(path, native_target, version, platform, bundle_package_type = :fmwk)
```

Comment:
@segiddins this method is copied pasted from `pod_target_installer.rb`, should we extract it out or make it a class method? The same question for the `update_changed_file` method below

Reply:
should probably stick both of them in a module and include that module in both places